assembly/src/library/namespace.rs

```diff
@@ -118,7 +126,7 @@ impl LibraryNamespace {
         if !source.starts_with(|c: char| c.is_ascii_lowercase() && c.is_ascii_alphabetic()) {
             return Err(LibraryNamespaceError::InvalidStart);
         }
-        if !source.chars().all(|c| c.is_ascii_alphanumeric() || matches!(c, '_')) {
+        if !source.chars().all(|c| c.is_ascii_graphic() || c.is_alphanumeric()) {
```

Comment:
If `Ident` didn't have an internal `SourceSpan` as discussed above, then we could use it in `LibraryNamespace` (and `ProcedureName`) to avoid duplicating the the "is ascii or alphanumeric" logic, right? Or am I missing some subtle reason we wouldn't want to do that?